tests_core/utils/model_zoo_tester.py

```diff
@@ -0,0 +1,136 @@
+        open(etag_file_path, 'w').write(etag_value)
+
+    @staticmethod
+    def _backup_old_version(model_dir):  # type: (str) -> None
+        if os.path.exists(model_dir):
+            backup_index = 0
+            while True:
+                dest = '{}.old.{}'.format(model_dir, backup_index)
+                if os.path.exists(dest):
+                    backup_index += 1
+                    continue
+                shutil.move(model_dir, dest)
+                break
+
+    def _prepare_model_data(self, model_test):  # type: (TestCase) -> Text
+        onnx_home = os.path.expanduser(os.getenv('ONNX_HOME', os.path.join('~', '.onnx')))
+        models_dir = os.getenv('ONNX_MODELS', os.path.join(onnx_home, 'models'))
+        model_dir = os.path.join(models_dir, model_test.model_name)  # type: Text
+        current_version_etag = self._get_etag_for_url(model_test.url)
+
+        # If model already exists, check if it's the latest version by verifying cached Etag value
+        if os.path.exists(os.path.join(model_dir, 'model.onnx')):
+            if not current_version_etag or current_version_etag == self._read_etag_file(model_dir):
+                return model_dir
+
+            # If model does exist, but is not current, backup directory
+            self._backup_old_version(model_dir)
+
+        # Download and extract model and data
+        download_file = tempfile.NamedTemporaryFile(delete=False)
+        temp_clean_dir = tempfile.mkdtemp()
+
+        try:
+            download_file.close()
+            print('\nStart downloading model {} from {}'.format(
+                model_test.model_name, model_test.url))
+            urlretrieve(model_test.url, download_file.name)
+            print('\nDone')
+
+            with tempfile.TemporaryDirectory() as temp_extract_dir:
+                with tarfile.open(download_file.name) as tar_file:
+                    tar_file.extractall(temp_extract_dir)
+
+                # Move model `.onnx` file from temp_extract_dir to temp_clean_dir
+                model_files = glob.glob(temp_extract_dir + '/**/*.onnx', recursive=True)
+                assert len(model_files) > 0, 'Model file not found for {}'.format(model_test.name)
+                model_file = model_files[0]
+                shutil.move(model_file, temp_clean_dir + '/model.onnx')
+
+                # Move extracted test data sets to temp_clean_dir
+                test_data_sets = glob.glob(temp_extract_dir + '/**/test_data_set_*', recursive=True)
+                test_data_sets.extend(
+                    glob.glob(temp_extract_dir + '/**/test_data_*.npz', recursive=True))
+                for test_data_set in test_data_sets:
+                    shutil.move(test_data_set, temp_clean_dir)
+
+                # Save Etag value to Etag file
+                self._write_etag_file(temp_clean_dir, current_version_etag)
+
+                # Move temp_clean_dir to ultimate destination
+                shutil.move(temp_clean_dir, model_dir)
```

Comment:
Why are you extracting `tar` archives to temporary directory and then move all extracted content to destination directory?  Why not just extract directly to destination directory?

Reply:
Some tar files contain additional files or a different file structure.
This gets around these problems.